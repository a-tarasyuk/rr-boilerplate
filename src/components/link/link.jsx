import React from 'react';
import PropTypes from 'prop-types';
import styles from './link.css';

export const Link = ({ href, name, title }) => (
  <a
    className={ styles.link }
    target={ 'blank' }
    href={ href }
    title={ title || name }
  >
    { name }
  </a>
);

Link.propTypes = {
  href: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  title: PropTypes.string,
};

Link.defaultProps = {
  title: '',
};

export default Link;
