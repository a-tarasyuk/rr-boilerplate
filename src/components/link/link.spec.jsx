import React from 'react';
import { shallow } from 'enzyme';
import { Link } from './link';

describe('<Link />', () => {
  it('matches its snapshot', () => {
    const component = shallow(
      <Link href={ 'http://localhost' } name={ 'Name' }>Link</Link>
    );
    expect(component).toMatchSnapshot();
  });
});
