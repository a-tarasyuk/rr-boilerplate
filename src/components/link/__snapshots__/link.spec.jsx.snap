// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> matches its snapshot 1`] = `
<a
  className="link"
  href="http://localhost"
  target="blank"
  title="Name"
>
  Name
</a>
`;
